src/baal/active/heuristics/heuristics.py

```diff
@@ -68,6 +69,31 @@ def wrapper(self, probabilities):
     return wrapper
 
 
+def gather_expand(data, dim, index):
+    """
+    Gather indices `index` from `data` after expanding along dimension `dim`.
+
+    Args:
+        data (tensor): A tensor of data.
+        dim (int): dimension to expand along.
+        index (tensor): tensor with the indices to gather.
+
+    Returns:
+        Tensor with the same shape as `index`.
+    """
+    max_shape = [max(dr, ir) for dr, ir in zip(data.shape, index.shape)]
```

Comment:
Could you add a reference to https://github.com/BlackHC/BatchBALD/blob/master/src/torch_utils.py ?